libs/designsystem/menu/src/menu.component.ts

```diff
@@ -75,33 +83,258 @@ export class MenuComponent implements AfterViewInit, OnDestroy {
+    const nextWordStartingWithChar = wordsStartingWithChar.filter(
+      (wordAndIndex) => wordAndIndex.index >= startIndex
+    )[0];
+
+    return nextWordStartingWithChar?.index ?? firstWordStartingWithChar.index;
+  }
+
+  focusItem() {
+    const itemToBeFocused = this.kirbyItems.get(this.focusedIndex);
+    const ionItem = itemToBeFocused.nativeElement.querySelector('ion-item');
+
+    // Look for interactive element within ion-item like toggle or checkbox and set focus if found
+    const firstInteractiveElementWithinItem = this.getFirstInteractiveElement(ionItem);
+    if (typeof firstInteractiveElementWithinItem?.['setFocus'] === 'function') {
+      firstInteractiveElementWithinItem['setFocus']();
+    } else {
+      this.focusSelectableItem(ionItem);
+    }
+  }
+
+  private focusSelectableItem(ionItem: HTMLIonItemElement) {
+    const nativeButton: HTMLButtonElement =
+      ionItem.shadowRoot.querySelector('button:not([disabled])');
+    nativeButton?.focus();
+  }
+
+  getTriggerButton(): HTMLButtonElement {
+    return (this.userProvidedButton ?? this.defaultButtonElement).nativeElement;
+  }
 
   public ngAfterViewInit(): void {
-    this.cdf.detectChanges(); // Sets the updated reference for kirby-floating
+    this.cdr.detectChanges(); // Sets the updated reference for kirby-floating
 
     this.zone.runOutsideAngular(() => {
       /*
        * Listen for ionScroll outside of Angular's change detection to
        * avoid a change detection cycle for every scroll-event fired
        */
       this.scrollListenerDisposeFn = this.renderer.listen(document, 'ionScroll', () => {
-        this.floatingDirective.hide();
+        this.floatingMenu.hide();
       });
     });
   }
 
+  ngAfterContentInit(): void {
+    this.setupAccessibilityForItems();
+    this.setupAccesibilityForUserProvidedButton();
+    this.setSelectableOnItems();
+  }
+
+  setSelectableOnItems() {
+    this.kirbyItemComponents.forEach((itemComponent) => {
+      if (itemComponent.selectable === undefined) {
+        itemComponent.selectable = true;
+      }
+    });
+  }
+
+  private setupAccessibilityForItems() {
+    this.kirbyItems.forEach((item) => {
+      this.setRoleAttributeForItem(item.nativeElement);
+    });
+  }
+
+  private setRoleAttributeForItem(item: HTMLElement) {
+    let menuItemRole = 'menuitem';
+    if (item.matches(':has(kirby-toggle, kirby-checkbox)')) {
+      menuItemRole = 'menuitemcheckbox';
+    } else if (item.matches(':has(kirby-radio)')) {
+      menuItemRole = 'menuitemradio';
+    }
+    this.renderer.setAttribute(item, 'role', menuItemRole);
+  }
+
+  menuVisibilityChanged(menuIsShown: boolean) {
+    this.floatingMenuIsShown = menuIsShown;
+    this.renderer.setAttribute(this.getTriggerButton(), 'aria-expanded', menuIsShown.toString());
+    if (!menuIsShown) {
+      this.focusedIndex = -1;
+      this.getTriggerButton().focus();
+    }
+  }
+
+  private setupAccesibilityForUserProvidedButton() {
```

Comment:
Consider more specific naming 
```suggestion
  private setUserProvidedButtonAriaAttributes() {
```